src/index.js

```diff
@@ -5,59 +5,63 @@ export default function ({ locals = {}, filters = {}, useMetadata = false } = {}
   const opts = arguments[0] || {}
 
   return (files, metalsmith, done) => {
-    setImmediate(done)
-
     // extend with metalsmith global metadata
     if (useMetadata) {
       locals = Object.assign({}, locals, metalsmith.metadata())
     }
 
-    Object.keys(files).forEach((file) => {
-      if (!/\.(pug|jade)/.test(path.extname(file))) {
-        return
-      }
+    try {
```

Comment:
Yeah that was my initial approach too.
Buuut, the nice thing about `throw` is that it break `forEach` like `break` does with a classic loop.
Hence I decided to wrap the whole block.
Otherwise we would need to accumulate errors in an array...